joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala

```diff
@@ -903,5 +903,39 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
       assocCallNode.argument.argumentIndex(1).head.code shouldBe "if"
       assocCallNode.argument.argumentIndex(2).head.code shouldBe "x.nil?"
     }
+
+    "have correct structure for proc definiton with procParameters and empty block" in {
+      val cpg =
+        code("-> (x,y) {}")
+      cpg.parameter.size shouldBe 3
+    }
+
+    "have correct structure for proc definiton with procParameters and non-empty block" in {
+      val cpg =
+        code("""-> (x,y) {
+            |if (x)
+            | y
+            |else
+            | b
+            |end
+            |}""".stripMargin)
+      cpg.parameter.size shouldBe 3
+      cpg.ifBlock.size shouldBe 1
+
+      cpg.parameter.head.code shouldBe "this"
```

Comment:
`this`? (Shouldn't it be `self`?) Either way, I reckon it would be useful to document this construction in the `astForProc..` method.